### PR TITLE
Use var lib for images and disks

### PR DESCRIPTION
**Pull request by tfussell**
Copied from https://github.com/giantswarm/k8s-kvm/pull/96

### CODE REVIEW

File: pkg/disk/disk.go [hunk 1]
```diff
@@ -33,7 +34,7 @@ func CreateDisks(guest *api.Guest) error {
 		gd := guest.Disks[i]
 
 		// set ID
-		gd.File = gd.ID + ".img"
+		gd.File = filepath.Join("/var/lib/containervmm/disks", gd.ID+".img")
```
Comment (tfussell) [inline pkg/disk/disk.go:37]: This allows us to mount this directory into the container so we can cache images.

Comment (tfussell) [inline pkg/disk/disk.go:37]: This allows us to mount a directory containing disks into the container so we can persist them and use host path volumes for them.

File: pkg/distro/flatcar.go [hunk 2]
```diff
@@ -283,64 +283,73 @@ ImM5rbOC6ZJdwLUTAg==
 
 // Pull Flatcar images from the official Kinvolk repository, optionally verify files and return the image names
 func DownloadImages(channel, version string, sanityChecks bool) (string, string, error) {
-	vmlinuzExistsLocal := util.FileExists(vmlinuz)
+	imageDirectory := filepath.Join("/var/lib/containervmm/images", channel, version)
```
Comment (tfussell) [inline pkg/distro/flatcar.go:288]: This allows us to cache the OS images by mounting a volume.

File: pkg/distro/flatcar.go [hunk 3]
```diff
@@ -283,64 +283,73 @@ ImM5rbOC6ZJdwLUTAg==
...
+	log.Infof("Downloading %s to %s", vmlinuzSignatureURL, vmlinuzSignaturePath)
 
-	if err := util.DownloadFile(vmlinuzSignature, vmlinuzSignatureURL); err != nil {
+	if err := util.DownloadFile(vmlinuzSignaturePath, vmlinuzSignatureURL); err != nil {
 		return fmt.Errorf("failed to download file from %s: %w", vmlinuzSignatureURL, err)
 	}
 
-	initrdSignatureURL := assetURL(channel, version, initrdSignature)
+	initrdSignaturePath := filepath.Join(imageDirectory, initrd+".sig")
+	initrdSignatureURL := assetURL(channel, version, initrd+".sig")
```
Comment (mazzy89) [inline pkg/distro/flatcar.go]:
```suggestion
	initrdSignatureURL := assetURL(channel, version, initrdSignature)
```

File: pkg/distro/flatcar.go [hunk 4]
```diff
@@ -283,64 +283,73 @@ ImM5rbOC6ZJdwLUTAg==
...
 	}
 
-	return vmlinuz, initrd, nil
+	return vmlinuzPath, initrdPath, nil
 }
 
-func downloadSignatures(channel, version string) error {
-	vmlinuzSignatureURL := assetURL(channel, version, vmlinuzSignature)
+func downloadSignatures(imageDirectory, channel, version string) error {
+	vmlinuzSignaturePath := filepath.Join(imageDirectory, vmlinuz+".sig")
```
Comment (mazzy89) [inline pkg/distro/flatcar.go]:
```suggestion
	vmlinuzSignaturePath := filepath.Join(imageDirectory, vmlinuzSignature)
```

Comment (mazzy89) [reply pkg/distro/flatcar.go]: just tiny change here to leave declared the full variable above.

File: pkg/distro/flatcar.go [hunk 5]
```diff
@@ -32,12 +34,10 @@ const (
 	archAMD64 = "amd64-usr"
 
 	// kernel
-	vmlinuz          = "flatcar_production_pxe.vmlinuz"
-	vmlinuzSignature = vmlinuz + ".sig"
+	vmlinuz = "flatcar_production_pxe.vmlinuz"
```
Comment (tfussell) [inline pkg/distro/flatcar.go]:
```suggestion
	vmlinuz          = "flatcar_production_pxe.vmlinuz"
	vmlinuzSignature = vmlinuz + ".sig"
```

File: pkg/distro/flatcar.go [hunk 6]
```diff
@@ -32,12 +34,10 @@ const (
...
 
 	// initrd
-	initrd          = "flatcar_production_pxe_image.cpio.gz"
-	initrdSignature = initrd + ".sig"
+	initrd = "flatcar_production_pxe_image.cpio.gz"
```
Comment (tfussell) [inline pkg/distro/flatcar.go]:
```suggestion
	initrd          = "flatcar_production_pxe_image.cpio.gz"
	initrdSignature = initrd + ".sig"
```

**Pull request closed without merging**